gcloud/monitoring/metric.py

```diff
@@ -0,0 +1,239 @@
+        self.type = type
+        self.labels = labels
+        self.metric_kind = metric_kind
+        self.value_type = value_type
+        self.unit = unit
+        self.description = description
+        self.display_name = display_name
+
+    @classmethod
+    def _fetch(cls, client, metric_type):
+        """Look up a metric descriptor by type.
+
+        :type client: :class:`gcloud.monitoring.client.Client`
+        :param client: The client to use.
+
+        :type metric_type: string
+        :param metric_type: The metric type name.
+
+        :rtype: :class:`MetricDescriptor`
+        :returns: The metric descriptor instance.
+
+        :raises: :class:`gcloud.exceptions.NotFound` if the metric descriptor
+            is not found.
+        """
+        path = '/projects/{project}/metricDescriptors/{type}'.format(
+            project=client.project,
+            type=metric_type)
+        info = client.connection.api_request(method='GET', path=path)
+        return cls._from_dict(info)
+
+    @classmethod
+    def _list(cls, client, filter_=None):
+        """List all metric descriptors for the project.
+
+        :type client: :class:`gcloud.monitoring.client.Client`
+        :param client: The client to use.
+
+        :type filter_: string or None
+        :param filter_: An optional filter string describing the metric
+            descriptors to be returned. See the `filter documentation`_.
+
+        :rtype: list of :class:`MetricDescriptor`
+        :returns: A list of metric descriptor instances.
+
+        .. _filter documentation:
+            https://cloud.google.com/monitoring/api/v3/filters
+        """
+        path = '/projects/{project}/metricDescriptors/'.format(
+            project=client.project)
+
+        descriptors = []
+
+        page_token = None
+        while True:
+            params = {}
+
+            if filter_ is not None:
+                params['filter'] = filter_
+
+            if page_token is not None:
+                params['pageToken'] = page_token
+
+            response = client.connection.api_request(
+                method='GET', path=path, query_params=params)
+            for info in response.get('metricDescriptors', ()):
+                descriptors.append(cls._from_dict(info))
+
+            page_token = response.get('nextPageToken')
+            if not page_token:
+                break
+
+        return descriptors
+
+    @classmethod
+    def _from_dict(cls, info):
+        """Construct a metric descriptor from the parsed JSON representation.
+
+        :type info: dict
+        :param info:
+            A ``dict`` parsed from the JSON wire-format representation.
+
+        :rtype: :class:`MetricDescriptor`
+        :returns: A metric descriptor.
+        """
+        return cls(
+            name=info['name'],
+            type=info['type'],
+            labels=tuple(LabelDescriptor._from_dict(label)
+                         for label in info.get('labels', [])),
```

Comment:
Missed this in https://github.com/GoogleCloudPlatform/gcloud-python/pull/1691/commits/41e8ff20c658ce1d29574a1770f20d11dc266188